Stack/Opc.Ua.Core/Security/Certificates/DirectoryCertificateStore.cs

```diff
@@ -377,40 +379,69 @@ public async Task<X509Certificate2> LoadPrivateKey(string thumbprint, string sub
                         };
 
-                        FileInfo privateKeyFile = new FileInfo(filePath.ToString() + ".pfx");
-                        if (!privateKeyFile.Exists)
+                        FileInfo privateKeyFilePfx = new FileInfo(filePath + ".pfx");
+                        FileInfo privateKeyFilePem = new FileInfo(filePath + ".pem");
+                        password = password ?? String.Empty;
+                        if (privateKeyFilePfx.Exists)
                         {
-                            Utils.LogError(Utils.TraceMasks.Security, "A private key for the certificate with thumbprint [{0}] does not exist.", certificate.Thumbprint);
-                            continue;
+                            certificateFound = true;
+                            foreach (var flag in storageFlags)
+                            {
+                                try
+                                {
+                                    certificate = new X509Certificate2(
+                                        privateKeyFilePfx.FullName,
+                                        password,
+                                        flag);
+                                    if (X509Utils.VerifyRSAKeyPair(certificate, certificate, true))
+                                    {
+                                        Utils.LogInfo(Utils.TraceMasks.Security, "Imported the PFX private key for [{0}].", certificate.Thumbprint);
+                                        return certificate;
+                                    }
+                                }
+                                catch (Exception ex)
+                                {
+                                    importException = ex;
+                                    certificate?.Dispose();
+                                }
+                            }
                         }
-
-                        certificateFound = true;
-                        password = password ?? String.Empty;
-                        foreach (var flag in storageFlags)
+                        // if PFX file doesn't exist, check for PEM file.
+                        else if (privateKeyFilePem.Exists)
                         {
+                            certificateFound = true;
                             try
                             {
-                                certificate = new X509Certificate2(
-                                    privateKeyFile.FullName,
-                                    password,
-                                    flag);
+                                if (m_pemResolverService == null || !m_pemResolverService.Active)
+                                {
+                                    byte[] pemDataBlob = File.ReadAllBytes(privateKeyFilePem.FullName);
+                                    certificate = CertificateFactory.CreateCertificateWithPEMPrivateKey(certificate, pemDataBlob, password);
+                                }
```

Comment:
@niveditha-sooda , I added an alternate path here to use the built in PEM loader.
Please check if the built in cross platform PEM loader could also do the job, then the IPemLoader static interface is not needed.

Reply:
The method CreateCertificateWithPEMPrivateKey has different implementations for .Net6 and netstandard2.0d for the latter case, it makes use of the Portable.BouncyCastle which we would want to avoid. We could discuss this in our upcoming meeting. Thanks.

Reply:
After discussion, it was decided that we remove the references for the IPemResolver and keep the changes as suggested by you. Thanks.